core/src/autogluon/core/metrics/__init__.py

```diff
@@ -374,7 +396,13 @@ def make_scorer(name, score_func, optimum=1, greater_is_better=True,
         cls = _QuantileScorer
     else:
         cls = _PredictScorer
-    return cls(name, score_func, optimum, sign, kwargs)
+    return cls(
+        name=name,
+        score_func=score_func,
+        optimum=optimum,
+        sign=sign,
+        kwargs=kwargs,
```

Comment:
Should we use `sign=sign, **kwargs` ?

Reply:
I actually updated the code just now so this works better and is fully generic: For example we don't want to fail when the metric needs `optimum` as a kwarg.

Reply:
You can refer to the new unit test to see how it works

Reply:
I see, this is the cached kwargs from the Scorer.